woodwork/deserialize.py

```diff
@@ -57,6 +59,43 @@ def _typing_information_to_woodwork_table(table_typing_info, validate, **kwargs)
     kwargs = loading_info.get('params', {})
     table_type = loading_info.get('table_type', 'pandas')
 
+    logical_types = {}
+    semantic_tags = {}
+    column_descriptions = {}
+    column_metadata = {}
+    use_standard_tags = {}
+    dtypes = {}
+    for col in table_typing_info['column_typing_info']:
+        col_name = col['name']
+
+        ltype_metadata = col['logical_type']
+        ltype = ww.type_system.str_to_logical_type(ltype_metadata['type'], params=ltype_metadata['parameters'])
+
+        tags = col['semantic_tags']
+
+        if 'index' in tags:
+            tags.remove('index')
+        elif 'time_index' in tags:
+            tags.remove('time_index')
+
+        logical_types[col_name] = ltype
+        semantic_tags[col_name] = tags
+        column_descriptions[col_name] = col['description']
+        column_metadata[col_name] = col['metadata']
+        use_standard_tags[col_name] = col['use_standard_tags']
+
+        if col['physical_type']['type'] == 'category':
+            # Make sure categories are recreated properly
+            cat_values = col['physical_type']['cat_values']
+            if table_type == 'pandas' and pd.__version__ > '1.1.5':
+                cat_object = pd.CategoricalDtype(pd.Index(cat_values, dtype='object'))
+            else:
+                cat_object = pd.CategoricalDtype(pd.Series(cat_values))
```

Comment:
Had to do this messy logic because of a difference in behavior between pandas 1.1.5 and 1.2.0 and greater. Since Koalas won't work with >1.1.5 yet, this was the only solution I found to allow tests to pass with Koalas using pandas <1.2.0 and other installs using pandas >=1.2.0.

Once Koalas removes their restriction, we can update our pandas min version and clean this up.

Reply:
Ah, well that's another reason to look forward to koalas fixing that restriction. Should we make an issue for this?